aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,576 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"log"
+	reflect "reflect"
+	"strings"
+	"testing"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_sagemaker_model", &resource.Sweeper{
+		Name: "aws_sagemaker_model",
+		F:    testSweepSagemakerModels,
+	})
+}
+
+func testSweepSagemakerModels(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).sagemakerconn
+
+	req := &sagemaker.ListModelsInput{
+		NameContains: aws.String("terraform-testacc-sagemaker-model"),
+	}
+	resp, err := conn.ListModels(req)
+	if err != nil {
+		return fmt.Errorf("Error listing models: %s", err)
+	}
+
+	if len(resp.Models) == 0 {
+		log.Print("[DEBUG] No sagemaker models to sweep")
+		return nil
+	}
+
+	for _, model := range resp.Models {
+		_, err := conn.DeleteModel(&sagemaker.DeleteModelInput{
+			ModelName: model.ModelName,
+		})
+		if err != nil {
+			return fmt.Errorf(
+				"Error deleting sagemaker model (%s): %s",
+				*model.ModelName, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSSagemakerModel_basic(t *testing.T) {
+	var model sagemaker.DescribeModelOutput
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model),
+					testAccCheckSagemakerModelName(&model, "terraform-testacc-sagemaker-model-foo"),
```

Comment:
Is there a reason these types of checks are here? It looks like you are confirming that the object looks correct but you're using that object to set the state and checking that [below](https://github.com/terraform-providers/terraform-provider-aws/pull/2478/files#diff-45a7fd3bdf025a4346f8ac9af2be156bR74). I'm not sure we need to double check like this

Reply:
Sounds reasonable. I looked at a lot of other tests, which seem to do these types of checks as well, for whatever reason (see https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_vpc_test.go#L104). I can remove `testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model)` as well, is that correct?

Reply:
It looks like those were using an older style of testing and we haven't had a chance to clean them up yet. If you look at the blame you can see that different people committed those test checks. For now, the best practice is just to use `resource.TestCheckResourceAttr`.